tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/TaggedMetricRegistry.java

```diff
@@ -38,6 +39,8 @@
      */
     Timer timer(MetricName metricName);
 
+    Timer timer(MetricName metricName, Supplier<Timer> timerSupplier);
```

Comment:
I don't know of any other implementations of `TaggedMetricRegistry` than `DefaultTaggedMetricRegistry` but we might want to implement these new methods via Java 8 `default`s here in interface for back compat

Reply:
I found one interface that extends `TaggedMetricRegistry` and one implementation of that extension, but I think we're ok to make this interface change without the `default` and just update that one usage.

Would still like to get test coverage over these new methods for hygiene, then happy to merge and tag a 0.10.0.